### PR TITLE
add ruby codegen

### PR DESCRIPTION
**Pull request by leonlnj**
## Description
To add Ruby codegen for UPI to support fluentd parser which is written in Ruby.

Buf is upgraded and some of the yaml syntax are changed. Custom remote plugins will [deprecated](https://buf.build/grpc/plugins). The plugin that can be ported are following the yaml syntax in the [curated plugins](https://buf.build/plugins).

Python plugin are not moved as the [dependencies](https://buf.build/jeffsawatzky/plugins/mypy-protobuf) are not migrated yet

## Key Changes
- `buf.gen.yaml` - added Ruby codegen plugin, modified yaml syntax for Go plugin
- `.github/workflows/ci.yaml` - upgrade buf to v1.11
- `.github/workflows/publish-ruby.yaml` - to publish proto to ruby gem (version will take from gemspec)

`caraml-upi-protos` v0.0.0 was published manually as part of testing. Before show CI trying to re-push (and expected rejection)
image

### CODE REVIEW

File: .github/workflows/publish-ruby.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      - name: Publish to RubyGems
+        run: |
+          mkdir -p $HOME/.gem
+          touch $HOME/.gem/credentials
+          chmod 0600 $HOME/.gem/credentials
+          printf -- "---\n:rubygems_api_key: ${GEM_HOST_API_KEY}\n" > $HOME/.gem/credentials
+          gem build *.gemspec
+          gem push *.gem
+        env:
+          GEM_HOST_API_KEY: "${{secrets.RUBYGEMS_API_KEY}}"
```
Comment (pradithya) [inline .github/workflows/publish-ruby.yaml:23]: Does the key belong to caraml-dev?

File: caraml-upi-protos.gemspec [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  s.summary     = "UPI generated proto"
+  s.description = "Universal Prediction Interface codegen proto"
+  s.authors     = ["Caraml Dev"]
+  s.email       = "caraml-dev@gojek.com"
+  s.license     = "Apache-2.0"
+  s.files       = Dir["gen/ruby/grpc/**/*.rb"]
+  s.homepage    = "https://github.com/caraml-dev/universal-prediction-interface/"
+  s.add_runtime_dependency "google-protobuf", ["~> 3.12"]
+  s.add_runtime_dependency "googleapis-common-protos-types", ["~> 1.5.0"]
+  s.add_runtime_dependency "ruby-protocol-buffers", ["~> 1.5.0"]
```
Comment (pradithya) [inline caraml-upi-protos.gemspec:13]: are both `ruby-protocol-buffers`  and `google-protobuf` required?

**Review by pradithya — APPROVED** [review]
Few comments, other than that LGTM

File: .github/workflows/publish-ruby.yaml (hunk 1 above)
Comment (leonlnj) [reply .github/workflows/publish-ruby.yaml:23]: yes caraml-dev

File: caraml-upi-protos.gemspec (hunk 2 above)
Comment (leonlnj) [reply caraml-upi-protos.gemspec]: ah `ruby-protocol-buffers` isn't required, will remove it

**Pull request merged**